reana/reana_benchmark/analyze.py

```diff
@@ -307,31 +295,35 @@ def _build_histogram_plot(
 
 def _build_total_time_histogram(
     df: pd.DataFrame, plot_parameters: Dict
-) -> (str, Figure):
+) -> Tuple[str, Figure]:
     title = plot_parameters["title"]
+    data = df[df["status"] == WorkflowStatus.finished]
+    series = data["runtime"] + data["pending_time"]
     return (
         "histogram_total_time",
-        _build_histogram_plot(
-            df["runtime"] + df["pending_time"], 10, "total_time", title
-        ),
+        _build_histogram_plot(series, 10, "total_time", title),
     )
 
 
-def _build_runtime_histogram(df: pd.DataFrame, plot_parameters: Dict) -> (str, Figure):
+def _build_runtime_histogram(
+    df: pd.DataFrame, plot_parameters: Dict
+) -> Tuple[str, Figure]:
     title = plot_parameters["title"]
+    data = df[df["status"] == WorkflowStatus.finished]["runtime"].dropna()
     return (
         "histogram_runtime",
-        _build_histogram_plot(df["runtime"], 10, "runtime", title),
+        _build_histogram_plot(data, 10, "runtime", title),
     )
 
 
 def _build_pending_time_histogram(
     df: pd.DataFrame, plot_parameters: Dict
-) -> (str, Figure):
+) -> Tuple[str, Figure]:
     title = plot_parameters["title"]
+    data = df["pending_time"].dropna()
```

Comment:
Shouldn't we filter `finished` workflows here as well?

Reply:
Depends, if you want to have only `finished/failed` or `running` in addition. When a workflow is in `running` status it already has defined `pending_time`. That is why I use `dropna()` here. But it may be confusing. Maybe, I should add some comments or even better filter it in the same way as it is done with `finished` so it will not create confusion later.

Reply:
I added filter so it more clear what statuses are include into `pending_time`

Reply:
I find it a bit confusing that other two histogram plots filters `finished` workflows and this one uses more statuses. Perhaps we should include this info in the title or in the legend?
So the only reason `dropna()` is useful is to filter out `failed` workflows which have no `pending_time`? That means it failed before starting? Or I am wrong?

Reply:
`dropna` filters `queued` and `pending`. `failed`, `running` and `finished` all have valid `pending_time` (meaning REANA provided exact start date for workflow) 

I can include it in the title. Not sure if this is important but always good to include.

Reply:
Done